homeassistant/components/influxdb.py

```diff
@@ -6,65 +6,70 @@
 """
 import logging
 
-import homeassistant.util as util
-from homeassistant.const import (EVENT_STATE_CHANGED, STATE_UNAVAILABLE,
-                                 STATE_UNKNOWN)
+import voluptuous as vol
+
+from homeassistant.const import (
+    EVENT_STATE_CHANGED, STATE_UNAVAILABLE, STATE_UNKNOWN, CONF_HOST,
+    CONF_PORT, CONF_SSL, CONF_VERIFY_SSL, CONF_USERNAME, CONF_BLACKLIST,
+    CONF_PASSWORD, CONF_WHITELIST)
 from homeassistant.helpers import state as state_helper
-from homeassistant.helpers import validate_config
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['influxdb==3.0.0']
 
 _LOGGER = logging.getLogger(__name__)
 
-DOMAIN = "influxdb"
-DEPENDENCIES = []
+CONF_DB_NAME = 'database'
+CONF_TAGS = 'tags'
 
+DEFAULT_DATABASE = 'home_assistant'
 DEFAULT_HOST = 'localhost'
 DEFAULT_PORT = 8086
-DEFAULT_DATABASE = 'home_assistant'
 DEFAULT_SSL = False
 DEFAULT_VERIFY_SSL = False
-
-REQUIREMENTS = ['influxdb==3.0.0']
-
-CONF_HOST = 'host'
-CONF_PORT = 'port'
-CONF_DB_NAME = 'database'
-CONF_USERNAME = 'username'
-CONF_PASSWORD = 'password'
-CONF_SSL = 'ssl'
-CONF_VERIFY_SSL = 'verify_ssl'
-CONF_BLACKLIST = 'blacklist'
-CONF_WHITELIST = 'whitelist'
-CONF_TAGS = 'tags'
+DOMAIN = 'influxdb'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_BLACKLIST, default=[]):
+            vol.All(cv.ensure_list, [cv.entity_id]),
+        vol.Optional(CONF_DB_NAME, default=DEFAULT_DATABASE): cv.string,
+        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_PORT, default=False): cv.boolean,
+        vol.Optional(CONF_SSL, default=False): cv.boolean,
+        vol.Optional(CONF_TAGS, default={}):
+            vol.All(cv.ensure_list, {cv.string: [cv.string]}),
```

Comment:
I doubt this can work. You first force it into a list, then validate it as a dict.